util/localpool/localpool_norace.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !race
+
+package localpool
+
+// Get gets an object from the pool.
+func (p *LocalPool) Get() interface{} {
+	pid := procPin()
+	slot := p.slots[pid]
+	objLen := len(slot.objs)
+	var result interface{}
+	if objLen > 0 {
+		result = slot.objs[objLen-1]
+		slot.objs = slot.objs[:objLen-1]
+		slot.getHit++
+	} else {
+		slot.getMiss++
+	}
+	procUnpin()
+	if result == nil {
+		result = p.newFn()
+	}
+	return result
+}
+
+// Put puts an object back to the pool.
+func (p *LocalPool) Put(obj interface{}) {
+	if p.resetFn != nil {
+		p.resetFn(obj)
+	}
+	pid := procPin()
+	slot := p.slots[pid]
+	if len(slot.objs) < p.sizePerProc {
+		slot.objs = append(slot.objs, obj)
+		slot.putHit++
+	} else {
+		slot.putMiss++
```

Comment:
We need a callback here to release resources hold by `obj`, for example, close channel.

Reply:
Objects that need to call destroy are not very common, so I added a bool return value for Put, the caller can call destroy manually if Put returns false.